spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesClientServicesFunction.java

```diff
@@ -40,8 +40,11 @@
  *
  * See KubernetesDiscoveryClientAutoConfiguration.servicesFunction
  *
+ * This is not used internally anymore and will be removed in a future release.
+ *
  * @author Georgios Andrianakis
  */
+@Deprecated(forRemoval = true)
```

Comment:
we don't need this now.

So initially, services where look-ed up only via this indirection, that it was OK having it.
Now, when I add support for ExternalName type of services, there will be _two_ places where services are being searched and this interface will only cause problems and inconsistencies. So I dropped it's usage and deprecated it.